### PR TITLE
Fix vote check

### PR DESCRIPTION
**Pull request by rkeene**
Fix supplied in #964 was incorrect, updated

### CODE REVIEW

**Comment by PlasmaPower** [comment]
I think we should get rid of that part entirely. Why's it necessary?

**Comment by rkeene** [comment]
I think we need to update the `vote_max()` interface to return a `vote_code` so we know if it came from the DB/cache or the current vote was inserted (vote) or not (replay)

File: rai/node/lmdb.cpp [hunk 1]
```diff
@@ -1884,15 +1885,24 @@ std::shared_ptr<rai::vote> rai::mdb_store::vote_generate (rai::transaction const
...
 
-std::shared_ptr<rai::vote> rai::mdb_store::vote_max (rai::transaction const & transaction_a, std::shared_ptr<rai::vote> vote_a)
+std::shared_ptr<rai::vote> rai::mdb_store::vote_max (rai::transaction const & transaction_a, std::shared_ptr<rai::vote> vote_a, rai::vote_code & vote_result)
 {
 	std::lock_guard<std::mutex> lock (cache_mutex);
 	auto current (vote_current (transaction_a, vote_a->account));
 	auto result (vote_a);
+
+
 	if (current != nullptr && current->sequence > result->sequence)
```
Comment (SergiySW) [inline rai/node/lmdb.cpp]: I guess this can be  
`current->sequence >= result->sequence`  
to prevent vote_code::vote result for current vote == new vote

Comment (rkeene) [reply rai/node/lmdb.cpp]: If current (DB/cache) is equal to the vote-in-question (result), we should yield "replay", right ?

Comment (SergiySW) [reply rai/node/lmdb.cpp]: I think so. Otherwise old solution with == was enough

**Comment by rkeene** [comment]
After looking into `node.active.vote()` (`rai::active_transactions::vote()`) more, I think just removing the consideration altogether is good.  It will return true if a replay *OR* the vote is being ignored because it is being too spammy (from `existing->election->vote()`, aka `rai::election::vote()`).

**Comment by rkeene** [comment]
The reason this code was correct before #964 was because it would always evaluate the second par of the `if (... || ...)` expression as false and thus irrelevant

**Pull request merged**